src/main/java/fr/openent/competences/service/impl/DefaultExportBulletinService.java

```diff
@@ -2237,6 +2248,14 @@ private void buildSousMatieres(JsonObject matiere, JsonArray tableauDeConversion
                     JsonObject moyenClasseSous = matiere.getJsonObject("_moyennesClasse");
                     Float moyCl = getMoyenneForSousMat(moyenClasseSous, idPeriode, idSousMat);
                     sousMat.put(MOYENNE_CLASSE, isNull(moyCl)? NN : moyCl);
+                    sousMat.put(Field.SUBCOEF,1);
+                    for(Service service : services){
+                        for(SubTopic subTopic : service.getSubtopics()){
+                            if(subTopic.getId().equals(sousMat.getInteger("id_type_sousmatiere"))){
+                                sousMat.put(Field.SUBCOEF,subTopic.getCoefficient());
+                            }
+                        }
+                    }
```

Comment:
Si je comprends bien, tu boucles pour chaque service, tu récupères une liste de subtopic et tu l'itères pour comparer avec ton id ?

Voici ma suggestion en utilisant les streams et flatMap : 
```suggestion
                   services.stream()
                            .map(Service::getSubtopics)
                            .flatMap(Collection::stream)
                            .forEach(subTopic -> {
                                if(subTopic.getId().equals(sousMat.getInteger("id_type_sousmatiere"))){
                                    sousMat.put(Field.SUBCOEF,subTopic.getCoefficient());
                                }
                            });
```

Tu streams ton services que tu vas mapper, ensuite tu flatMap chaque liste qu'il y a (donc le subtopic) et tu boucles dessus pour faire ta comparaison!

(si ca marche pas on voit ensemble mais l'idée est là)

Reply:
magic string "id_type_sousmatiere"